app/jobs/rollover_job.rb

```diff
@@ -3,5 +3,15 @@
 class RolloverJob < ApplicationJob
   queue_as :default
 
-  def perform(recruitment_cycle_id); end
+  BATCH_SIZE = 10
+
+  def perform
+    relation = RecruitmentCycle.current_recruitment_cycle.providers
+
+    BatchDelivery.new(relation:, stagger_over: 1.hour, batch_size: BATCH_SIZE).each do |batch_time, providers|
+      providers.pluck(:provider_code).each do |provider_code|
```

Comment:
I wonder if it would be cleaner to add a helper method to `ActiveRecord::Relation` or not? 🤔 

```suggestion
    RecruitmentCycle.current_recruitment_cycle.providers.in_batches_over_time(1.hour, batch_size: BATCH_SIZE) do |batch_time, providers|
      providers.pluck(:provider_code).each do |provider_code|
```

But I guess we land into the realm of monkey-patching again...

Reply:
> But I guess we land into the realm of monkey-patching again...
Yes, I'd avoid this path of adding objects to external libraries.

Wrapping and extending functionality is less intrusive and less error-prone.